### PR TITLE
Enforce increasing of the `Executor::VERSION` on each release

### PR DESCRIPTION
**Pull request by xgreenx**
Closes https://github.com/FuelLabs/fuel-core/issues/1835

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/services/upgradable-executor/src/executor.rs [hunk 1]
```diff
@@ -115,10 +115,22 @@ impl<S, R> Executor<S, R> {
     /// on the block, native execution is used. If the version is not the same
     /// as in the block, then the WASM executor is used.
     pub const VERSION: u32 = StateTransitionBytecodeVersion::MIN;
+
     /// This constant is used along with the `version_check` test.
     /// To avoid automatic bumping during release, the constant uses `-` instead of `.`.
-    #[cfg(test)]
-    pub const CRATE_VERSION: &'static str = "0-26-0";
+    /// Each release should have its own new version of the executor.
```
Comment (Voxelot) [inline crates/services/upgradable-executor/src/executor.rs:121]: we should document this in the `contributing.md` file to explain that this needs to be manually updated every time the state transition function changes.

File: crates/services/upgradable-executor/src/executor.rs [hunk 2]
```diff
@@ -685,18 +702,62 @@ mod test {
...
-        let executor_cate_version = Executor::<Storage, DisabledRelayer>::CRATE_VERSION
-            .to_string()
-            .replace('-', ".");
+        let dashed_crate_version = crate_version.to_string().replace('.', "-");
+        let mut seen_executor_versions =
+            BTreeSet::<StateTransitionBytecodeVersion>::new();
+        let seen_crate_versions = Executor::<Storage, DisabledRelayer>::CRATE_VERSIONS
+            .iter()
+            .map(|(crate_version, version)| {
+                let executor_cate_version = crate_version.to_string().replace('-', ".");
```
Comment (MitchTurner) [inline crates/services/upgradable-executor/src/executor.rs:715]:
```suggestion
                let executor_crate_version = crate_version.to_string().replace('-', ".");
```

File: crates/services/upgradable-executor/src/executor.rs [hunk 3]
```diff
@@ -685,18 +702,62 @@ mod test {
...
+                seen_executor_versions.insert(*version);
+                (executor_cate_version, *version)
```
Comment (MitchTurner) [inline crates/services/upgradable-executor/src/executor.rs:717]:
```suggestion
                (executor_crate_version, *version)
```

**Review by MitchTurner — DISMISSED** [review]
Approved with some minor spelling fixes.

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**